test/Concurrency/Runtime/async_task_withUnsafeCurrentTask.swift

```diff
@@ -0,0 +1,39 @@
+// RUN: %target-run-simple-swift( -Xfrontend -disable-availability-checking -parse-as-library) 2>&1 | %FileCheck %s --dump-input=always
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+// REQUIRES: swift_task_debug_log
```

Comment:
As I did not find a better way to test it... this depends on the SWIFT_TASK_DEBUG_LOG prints from ~AsyncTask